web/assets/js/app.js

```diff
@@ -1,372 +1,344 @@
+      ph: "Biogeokimia: pH",
+      spco2: "Biogeokimia: Tekanan karbon dioksida pada permukaan laut (Pa)",
+      no3: "Biogeokimia: Nitrat terlarut dalam air (mmol m<sup>-3</sup>)",
+      po4: "Biogeokimia: Fosfat terlarut dalam air (mmol m<sup>-3</sup>)",
+      si: "Biogeokimia: Silika terlarut dalam air (mmol m<sup>-3</sup>)",
+      fe: "Biogeokimia: Besi terlarut dalam air (mmol m<sup>-3</sup>)",
+      chl: "Biogeokimia: Total massa klorofil terlarut dalam air (mg m<sup>-3</sup>)",
+      phyc: "Biogeokimia: Konsentrasi fitoplankton terlarut dalam air (mmol m<sup>-3</sup>)",
+    };
+
+    for (const productId in datasets) {
+      for (const datasetId in datasets[productId]) {
+        for (const variable of datasets[productId][datasetId]) {
+          yield {
+            key: `cmems_${variable}`,
+            label: variableNames[variable],
+            productId,
+            datasetId,
+            variable,
+            sourceOptions: {
+              type: "raster",
+              tileSize: 256,
+              attribution:
+                "&copy; Copernicus Marine Environment Monitoring Service (CMEMS)",
+              tiles: [getCMEMSUrl(productId, datasetId, variable)],
+            },
+            layerOptions: {
+              type: "raster",
+              layout: {
+                visibility: "none",
+              },
             },
-          },
-        };
+          };
+        }
       }
     }
   }
-}
 
-var BASEMAPS = [
-  // ESRI
-  {
+  const BASEMAPS = [
     // source: https://leaflet-extras.github.io/leaflet-providers/preview/
     // source: https://www.arcgis.com/home/item.html?id=5ae9e138a17842688b0b79283a4353f6
     // alternative: https://docs.mapbox.com/mapbox-gl-js/example/style-ocean-depth-data/
-    key: "esri_ocean",
-    label: "ESRI World Ocean",
-    sourceOptions: {
-      type: "raster",
-      tiles: [
-        "https://server.arcgisonline.com/ArcGIS/rest/services/Ocean/World_Ocean_Base/MapServer/tile/{z}/{y}/{x}",
-      ],
-      tileSize: 256,
-      attribution:
-        "Tiles &copy; Esri &mdash; Sources: GEBCO, NOAA, CHS, OSU, UNH, CSUMB, National Geographic, DeLorme, NAVTEQ, and Esri",
-    },
-    layerOptions: {
-      type: "raster",
+    {
+      key: "esri_ocean",
+      label: "ESRI World Ocean",
+      sourceOptions: {
+        type: "raster",
+        tiles: [
+          "https://server.arcgisonline.com/ArcGIS/rest/services/Ocean/World_Ocean_Base/MapServer/tile/{z}/{y}/{x}",
+        ],
+        tileSize: 256,
+        attribution:
+          "Tiles &copy; Esri &mdash; Sources: GEBCO, NOAA, CHS, OSU, UNH, CSUMB, National Geographic, DeLorme, NAVTEQ, and Esri",
+      },
+      layerOptions: {
+        type: "raster",
+      },
     },
-  },
-
-  // CMEMS
-  ...getCMEMSSources(),
-];
-
-// ----------------------------------------
-// ALPINE APP
-// ----------------------------------------
-
-var map = null;
```

Comment:
The map object being declared here because it helps debugging the Mapbox API. Wrapping it inside Alpine data will make it virtually impossible to access.